src/cephadm/tests/test_util_funcs.py

```diff
@@ -0,0 +1,290 @@
+# Tests for various assorted utility functions found within cephadm
+#
+from unittest import mock
+
+import os
+
+import pytest
+
+from tests.fixtures import with_cephadm_ctx, import_cephadm
+
+_cephadm = import_cephadm()
+
+
+class TestCopyTree:
+    def _copy_tree(self, *args, **kwargs):
+        with with_cephadm_ctx([]) as ctx:
+            with mock.patch("cephadm.extract_uid_gid") as eug:
+                eug.return_value = (os.getuid(), os.getgid())
+                _cephadm.copy_tree(ctx, *args, **kwargs)
+
+    def test_one_dir(self, tmp_path):
+        """Copy one dir into a non-existing dest dir."""
+        src1 = tmp_path / "src1"
```

Comment:
It clearly seems to work, but would you mind explaining to me where `tmp_path` comes from? I can see it's a `PosixPath` object, just not sure how it's getting set.